components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java

```diff
@@ -177,6 +177,10 @@ public static Map<String, String> getPlaceholderValues(EmailTemplate emailTempla
                                     .substring(placeHolder.indexOf(".", placeHolder.indexOf("claim")) + 1));
                     if (StringUtils.isNotEmpty(userClaim)) {
                         placeHolderData.put(placeHolder, userClaim);
+                    } else {
+                        if (placeHolder.equals(NotificationConstants.EmailNotification.USER_CLAIM_PREFIX + ".givenname")) {
```

Comment:
For any claim if the claim is not there then we have to use empty right? Any specific reason to check for ".givenname"?

Reply:
updated by removing the given name check.